std/file.d

```diff
@@ -4966,28 +4953,49 @@ auto dirEntries(string path, SpanMode mode, bool followSymlink = true)
 }
 
 /// Duplicate functionality of D1's `std.file.listdir()`:
+version (StdDdoc)
 @safe unittest
 {
     string[] listdir(string pathname)
     {
-        import std.algorithm;
-        import std.array;
-        import std.file;
-        import std.path;
+        import std.algorithm.iteration : filter, map;
+        import std.array : array;
+        import std.path : baseName;
 
-        return std.file.dirEntries(pathname, SpanMode.shallow)
+        return dirEntries(pathname, SpanMode.shallow)
             .filter!(a => a.isFile)
-            .map!(a => std.path.baseName(a.name))
+            .map!(a => baseName(a.name))
             .array;
     }
 
-    void main(string[] args)
-    {
+    void main() {
         import std.stdio;
 
-        string[] files = listdir(args[1]);
-        writefln("%s", files);
-     }
+        string[] files = listdir("");
+        writeln(files);
+    }
+}
+
+/// Compile D files in parallel:
+version (StdDdoc)
+@system unittest
+{
+    import std.stdio;
+    import std.parallelism;
+    import std.process;
+
+    // Iterate over all *.d files in current directory and all its subdirectories
+    auto dFiles = dirEntries("", "*.d", SpanMode.depth);
+    foreach (d; dFiles)
+        writeln(d.name);
+
+    foreach (d; parallel(dFiles, 1)) // process each entry in a new thread
```

Comment:
```suggestion
    foreach (d; parallel(dFiles, 1)) // process each entry in a separate thread
```

`std.parallelism.parallel` uses a shared thread pool for the whole program, so there are at most `N+1` (IIRC) parallel invocations of `dmd`, where `N` is the number of CPU threads (e.g. 8, on a 4 core CPU with 2-way SMT). As was written before, the comment implied that there would be up to `dFiles.length` dmd processes in flight.